pos-frontend/src/components/SignUp.jsx

```diff
@@ -0,0 +1,223 @@
+                  </Grid>
+                  <Grid item xs={12} sm={6}>
+                    <TextField
+                      required
+                      fullWidth
+                      id="lastName"
+                      {...register("lastName", { required: "Lastname is required"})}
+                      error={!!errors.lastName}
+                      helperText={errors?.lastName ? errors.lastName.message : null}
+                      label="Last Name"
+                      name="lastName"
+                      autoComplete="family-name"
+                    />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <TextField
+                      required
+                      fullWidth
+                      id="email"
+                      label="Email Address"
+                      name="email"
+                      autoComplete="email"
+                      {...register("email", { 
+                        required: "Email is required", 
+                        pattern: { 
+                          value: /^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$/, 
+                          message: "Enter Valid email"
+                        } 
+                      })}
+                      error={!!errors.email}
+                      helperText={errors?.email ? errors.email.message : null}
+                    />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <TextField
+                      required
+                      fullWidth
+                      name="password"
+                      label="Password"
+                      type="password"
+                      id="password"
+                      autoComplete="new-password"
+                      {...register("password", { 
+                        required: "Password is required",
+                        pattern: {
+                          value: /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/,
+                          message: "Password should be of at least 8 characters including Capital, Small and Special character with at least one number"
+                        }
+                      })}
+                      error={!!errors.password}
+                      helperText={errors?.password ? errors.password.message : null}
+                    />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <TextField
+                      required
+                      fullWidth
+                      name="confirmPassword"
+                      label="Confirm Password"
+                      type="password"
+                      id="confirmPassword"
+                      {...register("confirmPassword", { 
+                        required: "Confirm password is required",
+                        validate: (val) => {
+                          if(watch('password') !== val) {
+                            return "Password does not match"
```

Comment:
define all messages in a common file and import from there ... change this in every place

Reply:
Under the utility folder, validationMessages.js is created.